package.json

```diff
@@ -14,12 +14,11 @@
   },
   "main": "./dist/index.cjs",
   "module": "./dist/index.js",
-  "types": "./dist/index.d.ts",
+  "types": "./dist/index.d.cts",
   "exports": {
-    "types": "./dist/index.d.ts",
```

Comment:
Doesn't this break ESM types? Where is `indes.d.ts` defined now?

I would expect following to work on both cases:

```json
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      },
      "require": {
        "types": "./index.d.cts",
        "default": "./index.cjs"
      }
    },
```

Reply:
In this case, TypeScript will automatically infer the location of the type declaration files because they match the file names/paths of the corresopnding JS files — see these sections of their docs for more info on this:

https://www.typescriptlang.org/docs/handbook/modules/reference.html#file-extension-substitution
https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports

The `types` field is really only necessary if the file paths/names do not match up (e.g., the type declaration file for `./dist/index.js` is located at `./dist/some-other-file.d.ts`).

In the case of this repository and its `tsup` build process, the JS files and type declaration files have matching names/paths, meaning these two `exports` objects are semantically equivalent:

```json
"exports": {
  ".": {
    "import": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "require": {
      "types": "./dist/index.d.cts",
      "default": "./dist/index.cjs"
    }
  }
}
```

```json
"exports": {
  ".": {
    "import": "./dist/index.js",
    "require": "./dist/index.cjs"
  }
}
```

We've been [using this pattern in our `tsup`-powered libraries](https://github.com/readmeio/oas/blob/780fbadf04cbc35214c81723a5352cb7c5368e86/packages/oas/package.json#L9-L47) for a while now and have been able to load them (complete with types) into both CJS and ESM repositories with no issues.

Happy to update this PR to explicitly specify these `types` fields if you'd prefer that! In our experience, this has proven to be unnecessary maintenance overhead than anything else.

Reply:
Nice, thanks for the links! Current change looks good to me as is.